client/components/advanced-credentials/index.tsx

```diff
@@ -89,17 +90,32 @@ const AdvancedCredentials: FunctionComponent< Props > = ( { action, host, role }
 	const { protocol } = credentials;
 	const isAtomic = hasCredentials && 'dynamic-ssh' === protocol;
 
+	const [ testCredentialsResult, setTestCredentialsResult ] = useState( false );
+	const [ testCredentialsLoading, setTestCredentialsLoading ] = useState( true );
+
+	useEffect( () => {
+		( async () => {
+			const results: { ok: boolean } = await wp.req.post( {
```

Comment:
These are usually worth to have some sort of error handling. We're already handling the success case, but what happens if the request fails?

Reply:
Thanks for the suggestions! Made changes as suggested!